qiskit/visualization/circuit/text.py

```diff
@@ -1133,6 +1133,14 @@ def add_connected_gate(node, gates, layer, current_cons, gate_wire_map):
                     layer.set_qubit(node.qargs[i], gate)
                     current_cons.append((actual_index, gate))
 
+        # AnnotatedOperation with ControlModifier
+        mod_control = None
+        if getattr(op, "modifiers", None):
+            for modifier in op.modifiers:
+                if isinstance(modifier, ControlModifier):
+                    mod_control = modifier
+                    break
+
```

Comment:
This would not work correctly when there are multiple `ControlModifier`s in the list. The simplest fix would be to use the function `_canonicalize_modifiers` from `annotated_operation.py` which would combine multiple control modifiers into one, as follows:

```
if getattr(op, "modifiers", None):
    canonical_modifiers = _canonicalize_modifiers(op.modifiers)
    for modifier in canonical_modifiers:
        if isinstance(modifier, ControlModifier):
            mod_control = modifier
            break
```